etna/datasets/tsdataset.py

```diff
@@ -132,13 +136,36 @@ def __init__(
         self.known_future = self._check_known_future(known_future, df_exog)
         self._regressors = copy(self.known_future)
 
+        self.hierarchical_structure = hierarchical_structure
+        self.current_df_level: Optional[str] = self._get_dataframe_level(df=self.df)
+        self.current_df_exog_level: Optional[str] = None
+
         if df_exog is not None:
             self.df_exog = df_exog.copy(deep=True)
             self.df_exog.index = pd.to_datetime(self.df_exog.index)
-            self.df = self._merge_exog(self.df)
+            self.current_df_exog_level = self._get_dataframe_level(df=self.df_exog)
+            if self.current_df_level == self.current_df_exog_level:
+                self.df = self._merge_exog(self.df)
```

Comment:
What do we do about this?
> If there is no hierarchy in the dataset this methods should fail in case of inconsistency between df and df_exog

Reply:
I guess we need to fix it in master, I will create a bug. What do you think?

Reply:
Ok, I agree, better to fix master.